docs/developer-docs/latest/plugins/upload.md

```diff
@@ -6,10 +6,47 @@ canonicalUrl: https://docs.strapi.io/developer-docs/latest/plugins/upload.html
 
 # Upload
 
-The `Upload` plugin is used to implement the Media Library available in the admin panel. With it you can upload any kind of file on your server or external providers such as **AWS S3**.
+The `Upload` plugin is the backend powering the [Media Library](../../../user-docs/latest/media-library/introduction-to-media-library.md) plugin available by default in the Strapi admin panel.
+
+Using either the Media Library UI or the upload API directly, you can upload any kind of file for use in your Strapi application.
+
+By default Strapi provides a [provider](../development/using-providers.md) that uploads files to a local directory. Additional providers are available should you want to upload your files to another location.
+
+The providers maintained by Strapi include:
+
+- [Amazon S3](https://www.npmjs.com/package/@strapi/provider-upload-aws-s3)
+- [Cloudinary](https://www.npmjs.com/package/@strapi/provider-upload-cloudinary)
+- [Local](https://www.npmjs.com/package/@strapi/provider-upload-local)
+- [Rackspace](https://www.npmjs.com/package/@strapi/provider-upload-rackspace)
 
 ## Configuration
 
+This section details configuration options for the default upload provider. If using another provider (e.g. AWS S3 or Rackspace), see the available configuration parameters in that provider's documentation.
```

Comment:
Shouldn't we help users a bit more? We could perhaps create a table with the other providers maintained by Strapi, and put a link to where users can find the information they need. I'm pretty sure we did something a bit like that in other parts of the doc. What do you think?

Reply:
Also, I think it could be useful to mention in that introduction what are the available configuration options, and then redirect from here to the sections below.

Reply:
I also had the same questioning ("what are these providers we're mentioning here?") and missed the list in the introduction 😄 

When you write "see the available configuration parameters in that provider's documentation", does that only include the Strapi-maintained providers, or also any other 3rd party provider that one can find on npm for instance? In the latter case, I think it would indeed be difficult to list all of them here.